File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    n = len(sequence)
    left, right = 0, 0
    current_sum = sequence[0]
    min_length = float('inf')
    answer = []

    while right < n:
        if current_sum == k:
            if (right - left) < min_length:
                min_length = right - left
                answer = [left, right]
            elif (right - left) == min_length and (not answer or left < answer[0]):
                answer = [left, right]

            current_sum -= sequence[left]
            left += 1

        if current_sum < k:
            right += 1
            if right < n:
                current_sum += sequence[right]
        elif current_sum > k:
            current_sum -= sequence[left]
            left += 1

    return answer

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int left=0,right=0;
        int sum=sequence[0];
        int minLen=Integer.MAX_VALUE;
        int[] answer=new int[] {-1,-1};
        
        while(right<sequence.length){
            if(sum==k){
                int len=right-left+1;
                if(len<minLen){
                    minLen=len;
                    answer[0]=left;
                    answer[1]=right;
                }
            }
            //0이 있을수도있으니까 앞에 0 제외시키기 (부등호들어가야함)
            if(sum>=k){
                sum-=sequence[left];
                left++;
            }
            else{
                right++;
                if(right<sequence.length){
                    sum+=sequence[right];
                }
            }
        }
        return answer;
        
        
    }
}